text/0000-rewards.md

```diff
@@ -0,0 +1,300 @@
+
+## Stakeholders
+
+We alter the validators rewards protocol, but with negligable impact upon rewards for honest validators who comply with hardware and bandwidth recommendations.  
+
+We shall still reward participation in relay chain concensus of course, which de facto means block production but not finality, but these current reward levels shall wind up greatly reduced.  Any validators who manipulate block rewards now could lose rewards here, simply because of rewards being shifted from block production to availability, but this sounds desirable.
+
+We've discussed roughly this rewards protocol in https://hackmd.io/@rgbPIkIdTwSICPuAq67Jbw/S1fHcvXSF and https://github.com/paritytech/polkadot-sdk/issues/1811 as well as related topics like https://github.com/paritytech/polkadot-sdk/issues/5122
+
+## Logic
+
+### Categories
+
+We alter the [current rewards scheme](https://wiki.polkadot.network/docs/maintain-guides-validator-payout) by reducing to roughly these proportions of total rewards:
+- 15-20% - Relay chain block production and uncle logic 
+- 5% - Anything else related to relay chain finality, primarily beefy proving, but maybe other tastes exist.
+- Any existing rewards for on-chain validity statements would only cover backers, so those rewards must be removed.
+
+We add roughly these proportions of total rewards covering parachain work:
+- 70-75% - approval and backing validity checks, with the backing rewards being required to be less than approval rewards.  
+- 5-10% - Availability redistribution from availability providers to approval checkers.  We do not reward for availability distribution from backers to availability providers.
+
+### Observation
+
+We track this data for each candidate during the approvals process:
+```
+/// Our subjective record of out availability transfers for this candidate.
+CandidateRewards {
+    /// Anyone who backed this parablock
+    backers: [AuthorityId; NumBackers],
+    /// Anyone to whome we think no-showed, even only briefly.
+    noshows: HashSet<AuthorityId>,
+    /// Anyone who sent us chunks for this candidate
+    downloaded_from: HashMap<AuthorityId,u16>,    
+    /// Anyone to whome we sent chunks for this candidate
+    uploaded_to: HashMap<AuthorityId,u16>,
+}
+```
+We no longer require this data during disputes.  
+<!-- You could optionally track a `downloaded_one: Option<AuthorityBitField>` too, for the nodes from whome we douwnloaded only one chunk, but this seems like premature optimization -->
+
+After we approve a relay chain block, then we collect all its `CandidateRewards` into an `ApprovalsTally`, with one `ApprovalTallyLine` for each validator.  In this, we compute `approval_usages` from the final run of the approvals loop, plus `0.8` for each backer.
```

Comment:
@burdges since the candidate reward is per candidate, and approvals tally is per validator, when you say:
> we compute `approval_usages` from the final run of the approvals loop, plus `0.8` for each backer.
here we should consider each backer of all candidates rewards?

Example:

```
candidates_rewards =
[
    {backers: [...; 3]},
    {backers: [...; 3]},
]

// approvals usage for a given relay block X
approvals_usage = {
    validatorA: 5, // counted 5 candidate approvals
    validatorB: 3, // counted 3 candidate approvals 
}
```
given the example, when computing the approval_usages + 0.8 for each backer we should consider the 6 backers?

Reply:
I placed 0.8 here simply so that backers get paid less than approval checkers, but 0.9 or 0.95 sound fine too.  It likely that 1 works because backers spend more than approval checkers.

I've now forgotten if I even considered the number of backers here.  We could count only the ones the RC sees.  A priori, we're maybe better off if we count them if we think they participate in availability, but I've not yet thought throught how that interacts with the tit-for-tat game.